lib/jsonapi/utils/support/pagination.rb

```diff
@@ -62,16 +62,21 @@ def record_count_for(records, options)
         #
         # @api private
         def paginator
-          @paginator ||= paginator_klass.new(page_params)
+          @paginator ||= begin
+            paginator_klass = "#{resource_paginator_name}_paginator".classify.constantize
```

Comment:
I changed this from being its own method to living inside `paginator` because a) it's not used anywhere else and b) I thought it might confusing to have `paginator`, `paginator_klass`, and `resource_paginator_name` methods.